src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.TranslationAction.cs

```diff
@@ -10,11 +10,27 @@ namespace Microsoft.CodeAnalysis;
 internal partial class SolutionCompilationState
 {
     /// <summary>
-    /// Represents a change that needs to be made to a <see cref="Compilation"/>, <see cref="GeneratorDriver"/>, or both in response to
-    /// some user edit.
+    /// Represents a change that needs to be made to a <see cref="Compilation"/>, <see cref="GeneratorDriver"/>, or both
+    /// in response to some user edit.
     /// </summary>
-    private abstract partial class CompilationAndGeneratorDriverTranslationAction
+    private abstract partial class TranslationAction
     {
+        /// <summary>
+        /// The original state of the project prior to the user edit.
+        /// </summary>
+        public readonly ProjectState OldProjectState;
+
+        /// <summary>
+        /// The state of the project after the user edit was made.
+        /// </summary>
+        public readonly ProjectState NewProjectState;
```

Comment:
adding both of these to the translation action simplified a lot of code.  we have code that would pass along the old-state and action, and code that compute the new-state and action.  Now, one abstraction works for all those cases.